src/formatting/expr.rs

```diff
@@ -236,10 +236,10 @@ pub(crate) fn format_expr(
             /* Retrieving the comments before and after cast */
             let prefix_span = mk_sp(
                 subexpr.span.hi(),
-                context.snippet_provider.span_before(expr.span, "as"),
+                context.snippet_provider.span_before_last(expr.span, "as") - BytePos(1),
```

Comment:
Think there's a separate bug in this util function that only backs the position off by one, but feels like it should be the length of the needle. Our needle is the `as` keyword, so moving it back by one to avoid snagging the `a` in the keyword